Update `Timeline` Model and Establish Relationship with `Node` Entity

HunorTotBagi opened the pull request:
In this PR I have:
- Created `TimelinebaseDto` and alongside with `TimelineDto` moved to `BuildingBlocks`
- Moved Timeline `Events` folder to `BuildingBlocks`
- Moved `TimelineId` to `BuildingBlocks`
- Connected `Timeline` and `Node` entities
- Fixed database seeding for nodes
- Added `Timelines` repository

# Current state:
### Node
- Create Node ✅ 
- Get Node by Id ✅ 
- List Nodes ✅ 
- Delete Node by Id ✅ 
### Timelines
- Create Timeline ✅ 
- Get Timeline by Id ✅ 
- List Timelines ✅ 
- Delete Timeline by Id ✅

NikolaDmitrasinovic commented on Backend/src/Modules/Nodes/Nodes.Application/Data/NodesService.cs:
```diff
@@ -14,7 +15,11 @@ public class NodesService(INodesRepository nodesRepository, IServiceProvider ser
         var node = await nodesRepository.GetNodeByIdAsync(nodeId, cancellationToken);
-        var nodeDto = node.ToNodeDto();
+
+        var timelineService = serviceProvider.GetRequiredService<ITimelinesService>();
```
I would rather use DI here, but that might require too many changes...

HunorTotBagi replied:
I am summoning @NikolaVetnic to decide on this one

NikolaVetnic replied:
@HunorTotBagi why did you do it like this in the first place?
@NikolaDmitrasinovic why might that require too many changes?

HunorTotBagi replied:
@NikolaVetnic the same reason you did it in the `ReminderServices` :)

NikolaVetnic replied:
We agreed to let it slide but to refactor it at a later point 🥴

NikolaDmitrasinovic commented on Backend/src/Modules/Nodes/Nodes.Application/Entities/Nodes/Commands/CreateNode/CreateNodeCommand.cs:
```diff
@@ -13,40 +23,40 @@ public class CreateNodeCommandValidator : AbstractValidator<CreateNodeCommand>
             .MaximumLength(500).WithMessage("Description must not exceed 500 characters.");
 
-        RuleFor(x => x.Node.Timestamp)
+        RuleFor(x => x.Timestamp)
```
`Timestamp` doesn't sound like something some1 could just type in... Maybe this should be checked. I would think that this type of field/prop is automatically set.

HunorTotBagi replied:
I am summoning @NikolaVetnic to decide on this one

HunorTotBagi commented on Backend/src/BuildingBlocks/BuildingBlocks.Domain/Timelines/Timeline/ValueObjects/TimelineId.cs:
```diff
@@ -0,0 +1,77 @@
+{
+    public TimelineIdValueConverter()
+        : base(
+            timelineId => timelineId.Value, // Convert from NodeId to Guid
+            guid => TimelineId.Of(guid)) // Convert from Guid to NodeId
```
That's just a mistake from a copy paste. Changes made.

HunorTotBagi commented:
@NikolaVetnic During work on [Update `File` Model and Establish Relationship with `Node` Entity](https://github.com/NikolaVetnic/Timelines/pull/75) I discovered that deleting a File did not remove its ID from the corresponding Nodes list. I have addressed this issue in the mentioned PR but I fixed it also here in the last two commits, so a re-review is necessary.

NikolaVetnic commented on Backend/src/Modules/Nodes/Nodes.Application/Entities/Nodes/Commands/DeleteNode/DeleteNodeHandler.cs:
```diff
@@ -1,9 +1,10 @@
 namespace Nodes.Application.Entities.Nodes.Commands.DeleteNode;
 
-public class DeleteNodeHandler(INodesDbContext dbContext) : ICommandHandler<DeleteNodeCommand, DeleteNodeResult>
+public class DeleteNodeHandler(INodesDbContext dbContext, ITimelinesService timelinesService) : ICommandHandler<DeleteNodeCommand, DeleteNodeResult>
```
Here you are mixing two approaches - using services and the DbContext directly. You should implement this logic for deleting a Node in a service, and then use the service here. Reason - deleting a node from elsewhere with this code would not work properly as this is only triggered by hitting the appropriate endpoint.

HunorTotBagi replied:
Changes made.

NikolaVetnic commented on Backend/src/Modules/Nodes/Nodes.Infrastructure/Data/Extensions/InitialData.cs:
```diff
@@ -15,7 +16,8 @@ internal static class InitialData
-                ["investigation", "interview"]
+                ["investigation", "interview"],
+                TimelineId.Of(Guid.Parse("d87c1c54-b287-4a82-b45b-65db1d71e2fd"))
             ),
```
We will have to create this using the constructor, check out what I did with the Timeline initial data.

HunorTotBagi replied:
As discussed on Teams, @NikolaVetnic will take a look into this since there is a problem with `Category` and `Tags` `readonly` lists

HunorTotBagi commented on Backend/src/Modules/Timelines/Timelines.Application/Data/TimelinesService.cs:
```diff
@@ -0,0 +1,53 @@
+    public async Task AddNode(TimelineId timelineId, NodeId nodeId, CancellationToken cancellationToken)
+    {
+        var timeline = await timelinesRepository.GetTimelineByIdAsync(timelineId, cancellationToken);
+        timeline.AddNode(nodeId);
```
When creating a new `Node`, you must provide the ID of a speЦific `Timeline`, which effectively addresses this. I have tested it, and the IDs are correctly assigned in the appropriate locations.

HunorTotBagi commented on Backend/src/Modules/Timelines/Timelines.Application/Data/TimelinesService.cs:
```diff
@@ -0,0 +1,53 @@
+    public async Task RemoveNode(TimelineId timelineId, NodeId nodeId, CancellationToken cancellationToken)
+    {
+        var timeline = await timelinesRepository.GetTimelineByIdAsync(timelineId, cancellationToken);
+        timeline.RemoveNode(nodeId);
```
When you are deleting an entire `Node` that node is gone, why would you want to remove a `Timeline` ID from a deleted `Node`?

HunorTotBagi commented on Backend/src/Modules/Timelines/Timelines.Application/Data/TimelinesService.cs:
```diff
@@ -0,0 +1,53 @@
+        {
+            var node = await nodesService.GetNodeBaseByIdAsync(nodeId, cancellationToken);
+            timelineDto.Nodes.Add(node);
+        }
```
Changes made.
